geode-for-redis/src/integrationTest/java/org/apache/geode/redis/internal/commands/executor/list/AbstractLTrimIntegrationTest.java

```diff
@@ -0,0 +1,228 @@
+
+  @Test
+  public void withNonExistentKey_returnsNull() {
+    assertThat(jedis.ltrim("nonexistent", 0, -1)).isEqualTo("OK");
+  }
+
+  @Test
+  public void withNonIntegerRangeSpecifier_Fails() {
+    jedis.lpush(KEY, "e1", "e2", "e3", "e4");
+
+    assertThatThrownBy(() -> jedis.sendCommand(KEY, Protocol.Command.LTRIM, KEY,
+        "0", "not-an-integer"))
+            .hasMessage(ERROR_NOT_INTEGER);
+    assertThatThrownBy(() -> jedis.sendCommand(KEY, Protocol.Command.LTRIM, KEY,
+        "not-an-integer", "-1"))
+            .hasMessage(ERROR_NOT_INTEGER);
+    assertThatThrownBy(() -> jedis.sendCommand(KEY, Protocol.Command.LTRIM, KEY,
+        "not-an-integer", "not-an-integer"))
+            .hasMessage(ERROR_NOT_INTEGER);
+  }
+
+  @Test
+  public void trimsToSpecifiedRange_givenValidRange() {
+    initializeTestList();
+
+    jedis.ltrim(KEY, 0, 0);
+    assertThat(jedis.llen(KEY)).isEqualTo(1);
+    assertThat(jedis.lindex(KEY, 0)).isEqualTo("e4");
+
+    initializeTestList();
+
+    jedis.ltrim(KEY, 0, 1);
+    assertThat(jedis.llen(KEY)).isEqualTo(2);
+    assertThat(jedis.lindex(KEY, 0)).isEqualTo("e4");
+    assertThat(jedis.lindex(KEY, 1)).isEqualTo("e3");
+
+    initializeTestList();
+
+    jedis.ltrim(KEY, 0, 2);
+    assertThat(jedis.llen(KEY)).isEqualTo(3);
+    assertThat(jedis.lindex(KEY, 0)).isEqualTo("e4");
+    assertThat(jedis.lindex(KEY, 1)).isEqualTo("e3");
+    assertThat(jedis.lindex(KEY, 2)).isEqualTo("e2");
+
+    initializeTestList();
+
+    jedis.ltrim(KEY, 1, 2);
+    assertThat(jedis.llen(KEY)).isEqualTo(2);
+    assertThat(jedis.lindex(KEY, 0)).isEqualTo("e3");
+    assertThat(jedis.lindex(KEY, 1)).isEqualTo("e2");
+
+    initializeTestList();
+
+    jedis.ltrim(KEY, 1, -1);
+    assertThat(jedis.llen(KEY)).isEqualTo(3);
+    assertThat(jedis.lindex(KEY, 0)).isEqualTo("e3");
+    assertThat(jedis.lindex(KEY, 1)).isEqualTo("e2");
+    assertThat(jedis.lindex(KEY, 2)).isEqualTo("e1");
+
+    initializeTestList();
+
+    jedis.ltrim(KEY, 1, -2);
+    assertThat(jedis.llen(KEY)).isEqualTo(2);
+    assertThat(jedis.lindex(KEY, 0)).isEqualTo("e3");
+    assertThat(jedis.lindex(KEY, 1)).isEqualTo("e2");
+
+    initializeTestList();
+
+    jedis.ltrim(KEY, -2, -1);
+    assertThat(jedis.llen(KEY)).isEqualTo(2);
+    assertThat(jedis.lindex(KEY, 0)).isEqualTo("e2");
+    assertThat(jedis.lindex(KEY, 1)).isEqualTo("e1");
+
+    initializeTestList();
+
+    jedis.ltrim(KEY, -1, -1);
+    assertThat(jedis.llen(KEY)).isEqualTo(1);
+    assertThat(jedis.lindex(KEY, 0)).isEqualTo("e1");
+  }
+
+  @Test
+  public void trimsToCorrectRange_givenSpecifiersOutsideListSize() {
+    initializeTestList();
+
+    jedis.ltrim(KEY, -4, -1);
+    assertThat(jedis.llen(KEY)).isEqualTo(4);
+    assertThat(jedis.lindex(KEY, 0)).isEqualTo("e4");
+    assertThat(jedis.lindex(KEY, 1)).isEqualTo("e3");
+    assertThat(jedis.lindex(KEY, 2)).isEqualTo("e2");
+    assertThat(jedis.lindex(KEY, 3)).isEqualTo("e1");
+
+    initializeTestList();
```

Comment:
I may be wrong, but I don't think it's necessary to do this method call between each of your test cases. You never actually modify the list

Reply:
In this test case, you're right.